ui/app/store/actions.js

```diff
@@ -415,23 +415,25 @@ export function showQrScanner () {
 }
 
 export function setCurrentCurrency (currencyCode) {
-  return (dispatch) => {
+  return async (dispatch) => {
     dispatch(showLoadingIndication())
     log.debug(`background.setCurrentCurrency`)
-    background.setCurrentCurrency(currencyCode, (err, data) => {
+    let data
+    try {
+      data = await promisifiedBackground.setCurrentCurrency(currencyCode)
+    } catch (error) {
       dispatch(hideLoadingIndication())
-      if (err) {
-        log.error(err.stack)
-        return dispatch(displayWarning(err.message))
-      }
-      dispatch({
-        type: actionConstants.SET_CURRENT_FIAT,
-        value: {
-          currentCurrency: data.currentCurrency,
-          conversionRate: data.conversionRate,
-          conversionDate: data.conversionDate,
-        },
-      })
+      log.error(error.stack)
+      return dispatch(displayWarning(error.message))
```

Comment:
Is this _supposed_ to be the return value?

Reply:
My goal with this PR was to minimize changes in behavior. The return value should be identical to before, except that now it's wrapped in a Promise.

I don't think this particular return value is very useful, but 🤷‍♂️. We can address that separately.

Reply:
Fair enough

Reply:
Though the action itself used to return `undefined` where it now returns `Promise<whateverThisLineReturns>`, no?

Reply:
Ah, right, good point 🤔. That value was being `return`-ed to nowhere beforehand.

I'll move the `return` to a separate line then. Good catch!

Reply:
This has been updated.